jerry-core/parser/js/js-scanner-util.c

```diff
@@ -225,6 +308,548 @@ scanner_seek (parser_context_t *context_p) /**< context */
+                                     &context_p->token.lit_location);
+} /* scanner_add_literal */
+
+/**
+ * Add the current literal token to the current literal pool and
+ * set SCANNER_LITERAL_NO_REG if it is inside a with statement.
+ *
+ * @return pointer to the literal
+ */
+inline void JERRY_ATTR_ALWAYS_INLINE
+scanner_add_reference (parser_context_t *context_p, /**< context */
+                       scanner_context_t *scanner_context_p) /**< scanner context */
+{
+  lexer_lit_location_t *lit_location_p = scanner_add_custom_literal (context_p,
+                                                                     scanner_context_p->active_literal_pool_p,
+                                                                     &context_p->token.lit_location);
+  if (scanner_context_p->active_literal_pool_p->status_flags & SCANNER_LITERAL_POOL_IN_WITH)
+  {
+    lit_location_p->type |= SCANNER_LITERAL_NO_REG;
+  }
+} /* scanner_add_literal */
+
+/**
+ * Append an argument to the literal pool. If the argument is already present, make it a "hole".
+ */
+void
+scanner_append_argument (parser_context_t *context_p, /**< context */
+                         scanner_context_t *scanner_context_p) /**< scanner context */
+{
+  scanner_literal_pool_t *literal_pool_p = scanner_context_p->active_literal_pool_p;
+  parser_list_iterator_t literal_iterator;
+  parser_list_iterator_init (&literal_pool_p->literal_pool, &literal_iterator);
+  lexer_lit_location_t *literal_p;
+
+  const uint8_t *char_p = context_p->token.lit_location.char_p;
+  prop_length_t length = context_p->token.lit_location.length;
+
+  if (JERRY_LIKELY (!context_p->token.lit_location.has_escape))
+  {
+    while ((literal_p = (lexer_lit_location_t *) parser_list_iterator_next (&literal_iterator)) != NULL)
+    {
+      if (literal_p->length == length)
+      {
+        if (JERRY_LIKELY (!literal_p->has_escape))
+        {
+          if (memcmp (literal_p->char_p, char_p, length) == 0)
+          {
+            literal_p->length = 0;
+            break;
+          }
+        }
+        else if (lexer_compare_identifiers (literal_p->char_p, char_p, length))
+        {
+          literal_p->length = 0;
+          break;
+        }
+      }
+    }
+  }
+  else
+  {
+    while ((literal_p = (lexer_lit_location_t *) parser_list_iterator_next (&literal_iterator)) != NULL)
+    {
+      if (literal_p->length == length
+          && lexer_compare_identifiers (literal_p->char_p, char_p, length))
+      {
+        literal_p->length = 0;
+        break;
+      }
+    }
+  }
+
+  literal_p = (lexer_lit_location_t *) parser_list_append (context_p, &literal_pool_p->literal_pool);
+  *literal_p = context_p->token.lit_location;
+
+  literal_p->type = SCANNER_LITERAL_IS_ARG;
+} /* scanner_append_argument */
+
+/**
+ * Check whether the last identifier is an "eval".
+ */
+void
+scanner_check_eval (parser_context_t *context_p, /**< context */
```

Comment:
IMHO the name is misleading: If it checks something then it should return something. This method checks the 'eval' literal but it also sets a status flag.

Reply:
What is your suggestion? As for me `is_eval` returns with something, but anything is good for me.

Reply:
What if `scanner_update_eval` ? Also the function's comment needs update, to detail that it'll check the eval identifier and update the flag.

Reply:
I changed it to `scanner_detect_eval_call`, what do you think?